### DISCUSSION TITLE
getting concordance_index_ipcw as score in grid search

### DISCUSSION THREAD

**Discussion by asalimih** [sebp/scikit-survival#156] [Q&A]
Hi,
how can I have the concordance_index_ipcw score in the cv_results_ in the following example:
```
skf = StratifiedKFold(n_splits=5, shuffle=True, random_state=RD_STATE_val)
gcv = GridSearchCV(
    CoxnetSurvivalAnalysis(l1_ratio=0.9),
    param_grid={"coxnetsurvivalanalysis__alphas": [[v] for v in coxnet.alphas_]},
    cv=skf,
    error_score=0.5,
    n_jobs=4).fit(x_wtrain, y_wtrain)

cv_results = pd.DataFrame(gcv.cv_results_)
```
Thanks for your great package

**Comment by asalimih** [comment]
I think I solved this:
```
from sklearn.metrics import make_scorer
def ci_uno(y_true, y_pred):
    return concordance_index_ipcw(y_true,y_true, y_pred)[0]
ci_uno_scorer = make_scorer(ci_uno)

skf = StratifiedKFold(n_splits=5, shuffle=True, random_state=RD_STATE_val)
gcv = GridSearchCV(
    make_pipeline(CoxnetSurvivalAnalysis(l1_ratio=0.9)),
    param_grid={"coxnetsurvivalanalysis__alphas": [[v] for v in coxnet.alphas_]},
    cv=skf,
    error_score=0.5,
    scoring=ci_uno_scorer,
    n_jobs=4).fit(x_wtrain, y_wtrain)

cv_results = pd.DataFrame(gcv.cv_results_)
```

**Reply by sebp** [reply]
If you don't pass a scoring function to GridSearchCV, the `score` function is used, which uses `concordance_index_ipcw` already. See also the question in issue #150.

**Reply by sebp** [reply]
Sorry, I just realized that my previous statement was wrong, the `score` method uses `concordance_index_censored`, **not** `concordance_index_ipcw`. You might want to try one of the suggestions [in this comment](https://github.com/sebp/scikit-survival/issues/150#issuecomment-738884853).

**Comment by xavier-orcutt** [comment]
Hi @asalimih, I'm interested in doing something similar with ```cumulative_dynamic_auc```. Does that code above work?

**Comment by sebp** [comment]
I'm currently working on that feature. It should land in master soon.
